untyped-lambda-calculus/article.md

```diff
@@ -0,0 +1,493 @@
+# Untyped Lambda Calculus
+
+When talking about Haskell, the term "lambda calculus" often crops up.
+It is a theoretical framework used to define the meaning of "computation" in many functional languages.
+Understanding lambda calculus can be very helpful when talking about Haskell programs.
+It is also used extensively in programming language theory research and functional language design.
+
+There are two broad kinds of lambda calculus: _untyped_ and _typed_.
```

Comment:
The introduction feels a bit too confused and a tad bit too long. 

Perhaps you want to just mention that it is at the bottom of many functional programming languages and used extensively, and add a "Where is lambda calculus used?" section at the end?

I think this would improve readability (and SEO as well 😅).

Reply:
I've cut down the intro a bit (arguably a lot), and moved the list of languages based on λ-calculus to the conclusion. Hopefully this is will work better for SEO :laughing: